src/main.rs

```diff
@@ -34,11 +36,18 @@ fn cli() -> clap::Command {
                 .arg(arg!(--"raw" "do not parse the clash"))
                 .arg(arg!(--"no-color" "don't use ANSI colors in the output"))
                 .arg(
-                    arg!(--"show-whitespace" [BOOL] "render ¶ and • in place of newlines and spaces (default: true)")
+                    arg!(-'w' --"show-whitespace" [BOOL] "render ¶ and • in place of newlines and spaces (default: true)")
                         .value_parser(clap::builder::BoolishValueParser::new())
+                        // Needed?
                         .default_missing_value("true")
                 )
                 .arg(arg!([PUBLIC_HANDLE] "hexadecimal handle of the clash"))
+                .arg(
+                    arg!(-'t' --"testcases" [TESTCASE_NUM] "show the inputs of the testset (shows all if no extra args)")
+                        .action(clap::ArgAction::Append)
+                        .value_parser(value_parser!(usize))
+                        .default_missing_value("999")
+                )
```

Comment:
This doesn't quite address #9 as it always prints the test case title and both the input and output which makes it unsuitable for piping into a solution.

However adding more flags to control which parts of the testcase you want to show gets messy if it's an argument for `clash show` which already has some incompatible flags (like --raw). For this reason I would prefer to have a separate `clash testcase` subcommand to do this, so you could use it like
```
clash testcase 1  # print testcase including title, input, and output
clash testcase --input 1  # only print the input, suitable for piping to a solution
clash testcase --output 1  # only print the output
```

I think `clash show` would benefit from this `-t` flag if it instead controlled *how many* examples you want to show with the regular output though, since some clashes (specifically reverse mode) often require looking at more than one.

Reply:
> I think clash show would benefit from this -t flag if it instead controlled how many examples you want to show with the regular output though, since some clashes (specifically reverse mode) often require looking at more than one.

By regular output do you mean the statement*? 

*I was just looking at CG, and they use "goal" for the first block of text and "statement" for the whole: goal, i/o sections, constraints etc. ... but also just "statement" for the first block of text (as shown in the help tooltip or even the sections themselves when you edit/create a clash)

So if I understood it correctly clash show -t would show the (let's say whole) statement, including the first i/o example atm PLUS some extra tests, according to some extra argument?

I'm not fully sold. If it was a reverse (that is not currently supported), you wouldn't want any of the whole statement, and we will need to have a separate function anyway. And if it wasn't reverse, it's probably handier to switch from watching only the whole statement to only a bunch of testcases (with the future clash testcase)?

Reply:
I meant that `clash show -t 3` would show three examples instead of just one. Specifying test cases by index is not super useful when you're showing the clash statement for the first time (because you wouldn't know which test case is which yet).